lib/SGListViewCell.js

```diff
@@ -50,20 +62,35 @@ var SGListViewCell = React.createClass({
     );
   },
 
+  /* Every time the element is laid out, it is responsible for removing its previous
+   * interval from the interval tree and registering a new interval:
+   */
   onLayout(evt) {
-    // When the cell has actually been layed out, record the rendered width & height
-    this.viewProperties.width = evt.nativeEvent.layout.width;
-    this.viewProperties.height = evt.nativeEvent.layout.height;
+    if( this.interval ) {
+      this.props.removeFromTree(this.interval)
+    }
+
+    var layout = evt.nativeEvent.layout
+
+    this.interval = [layout.y, layout.y + layout.height]
+    this.interval.cell = this
+
+    this.props.addToTree(this.interval)
+
+    this.viewProperties.width = layout.width;
+    this.viewProperties.height = layout.height;
+
+    // Set the initial visibility state the first time it's laid out:
+    if( ! this.initialized ) {
+      this.initialized = true
+      if( ! this.props.isVisible(this.interval) ) this.setVisibility(false)
```

Comment:
This line is pretty terse. Can we update with the style guide: brackets, spacing and logic annotation.